managed-ledger/src/main/java/org/apache/bookkeeper/mledger/LedgerOffloaderFactory.java

```diff
@@ -42,20 +42,39 @@
     boolean isDriverSupported(String driverName);
 
     /**
-     * Create a ledger offloader with the provided configuration, user-metadata and scheduler.
+     * Create a ledger offloader with the provided configuration, user-metadata, scheduler and offloaderStats.
      *
      * @param offloadPolicies offload policies
      * @param userMetadata user metadata
      * @param scheduler scheduler
      * @return the offloader instance
      * @throws IOException when fail to create an offloader
      */
+    default T create(OffloadPoliciesImpl offloadPolicies,
+             Map<String, String> userMetadata,
+             OrderedScheduler scheduler)
+            throws IOException {
+        return create(offloadPolicies, userMetadata, scheduler, null);
```

Comment:
After this change, Is it possible to throw a NullPointerEx using the variable `offloaderStats`?

Reply:
Thanks for your reminder. Here we shouldn't make the method default. This method won't invoke in pulsar directly. In the pulsar master branch, it only invokes the method with the param `offloaderStats`. This method is compatible with the nar package, let the user implements by themselves.